libraries/AP_Filesystem/examples/File_IO/File_IO.cpp

```diff
@@ -0,0 +1,100 @@
+//Simple File I/O example
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Filesystem/AP_Filesystem.h>
+#include <stdio.h>
+
+//set according to the data size in file
+#define BUFFER_SIZE 50
+
+const AP_HAL::HAL& hal = AP_HAL::get_HAL();
+
+void setup();
+void loop();
+
+int file_count = 0;
+char write_buf[BUFFER_SIZE];
+char read_buf[BUFFER_SIZE];
+const char *dir_path = nullptr;
+static void write_file(const char *);
+static void read_file(const char *);
+
+void setup()
+{
+    //creating a test directory to store all files
+    dir_path = "./test_dir";
+    hal.console->printf("Creating directory - %s\n", dir_path);
+    int ret = AP::FS().mkdir(dir_path);
+    if (ret == -1) {
+      if (errno == EEXIST) {
+        hal.console->printf("Directory - %s already exist\n", dir_path);
+      }
+      else {
+        hal.console->printf("Failed to create directory %s - %s\n", dir_path, strerror(errno));
+      }
+    }
+}
+
+void loop()
+{
+    //creating 10 files in test directory
+    if (file_count == 10) {
+      return;
```

Comment:
Would be nicer to print a message, delay, *then* return.  Some of the other examples do that.